lib/flecks/queue.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class Flecks::Queue
+	def initialize
+		@queue = ::Queue.new
+		@count = 0
+	end
+
+	def push(id, task, content)
+		@count += 1
+
+		Async do
+			@queue.push([id, task.call, content])
+		end
+	end
+
+	def drain
+		while @count > 0
+			yield(*@queue.pop)
+			@count -= 1
+		end
+	end
```

Comment:
We need to know when the queue is exhausted so we don’t wait forever so this yields while the count is greater than 1. Since we increment the count synchronously and since this part is single-threaded, this is correct.